src/features/impl_rkyv.rs

```diff
@@ -166,27 +167,36 @@ mod u64 {
 
 #[cfg(test)]
 mod test {
-    pub type DefaultSerializer = rkyv::ser::serializers::CoreSerializer<256, 256>;
-    pub type DefaultDeserializer = rkyv::Infallible;
-    use rkyv::ser::Serializer;
-    use rkyv::*;
+    /// The serializer type expected by [`rkyv::to_bytes()`].
+    pub type TestSerializer<'a> = rkyv::api::high::HighSerializer<
+        rkyv::util::AlignedVec,
+        rkyv::ser::allocator::ArenaHandle<'a>,
+        rkyv::rancor::Panic,
+    >;
+    /// The deserializer type expected by [`rkyv::deserialize()`].
+    pub type TestDeserializer = rkyv::api::high::HighDeserializer<rkyv::rancor::Panic>;
+    pub type TestValidator<'a> = rkyv::api::high::HighValidator<'a, rkyv::rancor::Panic>;
     pub fn test_archive<T>(value: &T)
     where
-        T: core::fmt::Debug + PartialEq + rkyv::Serialize<DefaultSerializer>,
-        T::Archived: core::fmt::Debug + PartialEq<T> + rkyv::Deserialize<T, DefaultDeserializer>,
+        T: core::fmt::Debug
+            + PartialEq
+            + rkyv::Portable
+            + for<'a> rkyv::Serialize<TestSerializer<'a>>,
+        T::Archived: core::fmt::Debug + PartialEq<T> + rkyv::Deserialize<T, TestDeserializer>,
+        // #[cfg(feature = "bytecheck")]
+        T::Archived: for<'a> rkyv::bytecheck::CheckBytes<TestValidator<'a>>,
     {
-        let mut serializer = DefaultSerializer::default();
-        serializer
-            .serialize_value(value)
-            .expect("failed to archive value");
-        let len = serializer.pos();
-        let buffer = serializer.into_serializer().into_inner();
-
-        let archived_value = unsafe { rkyv::archived_root::<T>(&buffer[0..len]) };
+        let buffer = rkyv::to_bytes(value).unwrap();
```

Comment:
The above was basically open-coding `rkyv::api::high::to_bytes()`, which we can easily inline.